### PR TITLE
initial release PR

### PR DESCRIPTION
**Pull request by dishiinboxru**
initial release PR , testing if all is visible

### CODE REVIEW

File: src/main/java/net/javaguides/springmvc/entity/Contact.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	@Column(name="name")
+	private String name;
+	
+	@Column(name="phone")
+	private String phone;
+	
+	public Contact() {
+	}
+
+	public int getId() {
```
Comment (L-Evg) [inline src/main/java/net/javaguides/springmvc/entity/Contact.java:29]: Lombok? https://projectlombok.org
https://www.baeldung.com/intro-to-project-lombok

File: src/main/java/net/javaguides/springmvc/service/ContactService.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+	List<Contact> getContacts();
+
+	Contact saveContact(Contact theContact);
+
+	Contact getContact(int theId) throws ResourceNotFoundException;
+
+	Contact getContactByName(String name) throws ResourceNotFoundException;
+
+
```
Comment (L-Evg) [inline src/main/java/net/javaguides/springmvc/service/ContactService.java:18]: Redundant blank line.

File: src/main/webapp/WEB-INF/views/contact-form.jsp [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<%@ page language="java" contentType="text/html; charset=ISO-8859-1" pageEncoding="ISO-8859-1" %>
+<%@ taglib uri="http://www.springframework.org/tags/form" prefix="form" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="ISO-8859-1">
```
Comment (L-Evg) [inline src/main/webapp/WEB-INF/views/contact-form.jsp:7]: Why not UTF-8?

File: src/main/webapp/WEB-INF/views/contact-form.jsp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    <title>Spring MVC 5 - form handling | Java Guides</title>
```
Comment (L-Evg) [inline src/main/webapp/WEB-INF/views/contact-form.jsp:8]: Hmmm... .

File: src/main/java/net/javaguides/springmvc/service/ContactService.java (hunk 2 above)
Comment (dishiinboxru) [reply src/main/java/net/javaguides/springmvc/service/ContactService.java:18]: fixed

File: src/main/webapp/WEB-INF/views/contact-form.jsp (hunk 3 above)
Comment (dishiinboxru) [reply src/main/webapp/WEB-INF/views/contact-form.jsp:7]: I haven't seen much difference before, but after digging a bit deeper looks like UTF-8 is more future-proof, so yes, I'll change that.

File: src/main/webapp/WEB-INF/views/contact-form.jsp (hunk 4 above)
Comment (dishiinboxru) [reply src/main/webapp/WEB-INF/views/contact-form.jsp:8]: updated

File: src/main/java/net/javaguides/springmvc/entity/Contact.java (hunk 1 above)
Comment (dishiinboxru) [reply src/main/java/net/javaguides/springmvc/entity/Contact.java:29]: Today - lombok, tomorrow - kotlin ?! ))
Ok, will do

File: README.md [hunk 5]
```diff
@@ -1,2 +1,20 @@
 # SpringForQA
```
Comment (L-Evg) [inline README.md:1]: .DS_Store - please remove the binary file from Git.

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    ports:
+      - '23306:3306'
+    # Set mysql environments
+    environment:
+      MYSQL_ROOT_PASSWORD: 'secret'
+      MYSQL_USER: 'user'
+      MYSQL_PASSWORD: 'pass'
+      MYSQL_DATABASE: 'test'
+    volumes:
+      - vikta-mysql-data-volume:/var/lib/mysql
```
[the remark this answers is not in the document]
Comment (L-Evg) [reply docker-compose.yml:16]: .

File: src/main/java/net/javaguides/springmvc/advice/ResourceNotFoundAdvice.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+@ControllerAdvice
+public class ResourceNotFoundAdvice {
+
+    @ResponseBody
+    @ExceptionHandler(ResourceNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    String resourceNotFoundHandler(ResourceNotFoundException ex) {
+        return String.format("resource not found '%s'", ex.getMessage());
+    }
+
```
Comment (L-Evg) [inline src/main/java/net/javaguides/springmvc/advice/ResourceNotFoundAdvice.java:19]: Redundant blank line.

File: src/main/java/net/javaguides/springmvc/config/AppInitializer.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    protected Class<?>[] getServletConfigClasses() {
+        return new Class[]{WebMvcConfig.class};
+    }
+
+    @Override
+    protected String[] getServletMappings() {
+        return new String[]{"/"};
+    }
+
+
```
Comment (L-Evg) [inline src/main/java/net/javaguides/springmvc/config/AppInitializer.java:22]: Redundant blank line.

File: src/main/java/net/javaguides/springmvc/config/PersistenceJPAConfig.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        final HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+        entityManagerFactoryBean.setJpaVendorAdapter(vendorAdapter);
+        entityManagerFactoryBean.setJpaProperties(additionalProperties());
+
+        return entityManagerFactoryBean;
+    }
+
+    final Properties additionalProperties() {
+        final Properties hibernateProperties = new Properties();
+        hibernateProperties.setProperty("hibernate.hbm2ddl.auto", env.getProperty("hibernate.hbm2ddl.auto"));
```
Comment (L-Evg) [inline src/main/java/net/javaguides/springmvc/config/PersistenceJPAConfig.java:52]: Looks a bit overhead. I believe there are more elegant ways to work with properties in Spring.

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,133 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>net.javaguides.springmvc</groupId>
+    <artifactId>springmvc5-example</artifactId>
+    <packaging>war</packaging>
+    <version>0.0.1-SNAPSHOT</version>
+    <name>SpringMVC5-SpringDataJPA2-JSP-MySQL-Example Maven Webapp</name>
+    <url>http://maven.apache.org</url>
```
[the remark this answers is not in the document]
Comment (dishiinboxru) [reply pom.xml:10]: updated

File: README.md [hunk 11]
```diff
@@ -1,2 +1,20 @@
...
+###prerequisites 
+
+    * Tomcat server 9.0.31 or higher
+    * docker-compose is available
+    * MySQl 5.0 or higher is available
+    * Java 1.8
+
+1 - after downloading/cloning the repo - perform docker-compose up in order to enable DB for the app
+
+2 - In IDEA Edit configuration in the following way : Use Tomcat Server 9.0 , local deployment .
```
[the remark this answers is not in the document]
Comment (dishiinboxru) [reply README.md:15]: ok, need to figure out the workaround.

File: docker-compose.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (dishiinboxru) [reply docker-compose.yml:16]: Frankly speaking, I didn't know if that  mattered at all , I assumed it was just some type of MySQL deployment in container. Googling "vikta with docker" , "vikta with MySQL" didn't return anything relevant. Could you please advise some further reading on that ?

File: src/main/java/net/javaguides/springmvc/advice/ResourceNotFoundAdvice.java (hunk 7 above)
Comment (dishiinboxru) [reply src/main/java/net/javaguides/springmvc/advice/ResourceNotFoundAdvice.java:19]: fixed

File: src/main/java/net/javaguides/springmvc/config/AppInitializer.java (hunk 8 above)
Comment (dishiinboxru) [reply src/main/java/net/javaguides/springmvc/config/AppInitializer.java:22]: fixed

File: README.md (hunk 5 above)
Comment (dishiinboxru) [reply README.md:1]: done

File: src/main/java/net/javaguides/springmvc/config/PersistenceJPAConfig.java (hunk 9 above)
Comment (dishiinboxru) [reply src/main/java/net/javaguides/springmvc/config/PersistenceJPAConfig.java]: eventually I've managed to make it like that - created a separate properties file for hibernate and added annotation 
@PropertySource({ "classpath:hibernate.properties" })
worked like a charm.

**Pull request merged**